File: batchbhavna/typesofAdapter/app/src/main/java/com/example/typesofadapter/RecyclerActivity.java
package com.example.typesofadapter;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.widget.LinearLayout;

import com.example.typesofadapter.adapters.RecyclerviewAdapter;
import com.example.typesofadapter.models.Items;

import java.util.ArrayList;

public class RecyclerActivity extends AppCompatActivity {
RecyclerView recyclerView;
ArrayList<Items> userlist;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_recycler);
        recyclerView = findViewById(R.id.recycleview);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getApplicationContext());
        recyclerView.setLayoutManager(linearLayoutManager);
        userlist = new ArrayList<>();



        userlist.add(new Items(R.drawable.arebic,"Android"));
        userlist.add(new Items(R.drawable.sample,"Sample"));
        userlist.add(new Items(R.drawable.right,"Right"));
        userlist.add(new Items(R.drawable.third,"Third"));
        userlist.add(new Items(R.drawable.arebic,"Android"));
        userlist.add(new Items(R.drawable.sample,"Sample"));
        userlist.add(new Items(R.drawable.right,"Right"));
        userlist.add(new Items(R.drawable.third,"Third"));
        userlist.add(new Items(R.drawable.arebic,"Android"));
        userlist.add(new Items(R.drawable.sample,"Sample"));
        userlist.add(new Items(R.drawable.right,"Right"));
        userlist.add(new Items(R.drawable.third,"Third"));
        userlist.add(new Items(R.drawable.arebic,"Android"));
        userlist.add(new Items(R.drawable.sample,"Sample"));
        userlist.add(new Items(R.drawable.right,"Right"));
        userlist.add(new Items(R.drawable.third,"Third"));
        userlist.add(new Items(R.drawable.arebic,"Android"));
        userlist.add(new Items(R.drawable.sample,"Sample"));
        userlist.add(new Items(R.drawable.right,"Right"));
        userlist.add(new Items(R.drawable.third,"Third"));
        userlist.add(new Items(R.drawable.arebic,"Android"));
        userlist.add(new Items(R.drawable.sample,"Sample"));
        userlist.add(new Items(R.drawable.right,"Right"));
        userlist.add(new Items(R.drawable.third,"Third"));

        RecyclerviewAdapter recyclerviewAdapter = new RecyclerviewAdapter(userlist,getApplicationContext());

        recyclerView.setAdapter(recyclerviewAdapter);
    }
}

File: 4pmclass/TypesofAdapters/app/src/main/java/com/example/typesofadapters/RecyclerActivity.kt
package com.example.typesofadapters

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.typesofadapters.apdatper.RecyclerAdapter
import com.example.typesofadapters.modelclass.RecyclerMode

class RecyclerActivity : AppCompatActivity() {
    lateinit var recylclerview:RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_recycler)
        var imagelist: MutableList<RecyclerMode>  = ArrayList<RecyclerMode>()
        recylclerview = findViewById(R.id.recyclerview)

        var layoutmanager = LinearLayoutManager(applicationContext)
        recylclerview.layoutManager = layoutmanager

        imagelist.add(RecyclerMode(R.drawable.download))
        imagelist.add(RecyclerMode(R.drawable.right))
        imagelist.add(RecyclerMode(R.drawable.naturetwo))
        imagelist.add(RecyclerMode(R.drawable.sample))
        imagelist.add(RecyclerMode(R.drawable.third))
        imagelist.add(RecyclerMode(R.drawable.weather))
        imagelist.add(RecyclerMode(R.drawable.download))
        imagelist.add(RecyclerMode(R.drawable.right))
        imagelist.add(RecyclerMode(R.drawable.naturetwo))
        imagelist.add(RecyclerMode(R.drawable.sample))
        imagelist.add(RecyclerMode(R.drawable.third))
        imagelist.add(RecyclerMode(R.drawable.weather))
        imagelist.add(RecyclerMode(R.drawable.download))
        imagelist.add(RecyclerMode(R.drawable.right))
        imagelist.add(RecyclerMode(R.drawable.naturetwo))
        imagelist.add(RecyclerMode(R.drawable.sample))
        imagelist.add(RecyclerMode(R.drawable.third))
        imagelist.add(RecyclerMode(R.drawable.weather))

        var reycyleradatper = RecyclerAdapter(imagelist,applicationContext)
        recylclerview.adapter = reycyleradatper



    }
}